app/src/App.tsx

```diff
@@ -1,36 +1,63 @@
 import { configureStore } from '@reduxjs/toolkit'
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { Provider } from 'react-redux'
 import { ThemeProvider } from 'styled-components'
 import Web3Provider from 'web3-react'
 
 import 'react-datepicker/dist/react-datepicker.css'
 import 'sanitize.css'
 
+import { MAINNET_LOCATION, XDAI_LOCATION } from './common/constants'
+import { HeaderNoRouter } from './components/common/layout/header'
 import { Main } from './components/main'
+import SettingsViewContainer from './components/settings/settings_view'
 import { ApolloProviderWrapper } from './contexts/Apollo'
 import { ConnectedCPK, ConnectedWeb3 } from './hooks'
 import balanceReducer from './store/reducer'
 import theme from './theme'
 import { GlobalStyle } from './theme/global_style'
 import connectors from './util/connectors'
+import { getInfuraUrl, networkIds } from './util/networks'
+import { checkRpcStatus, getNetworkFromChain } from './util/tools'
 
 const store = configureStore({ reducer: balanceReducer })
 
 const App: React.FC = () => {
+  const windowObj: any = window
+
+  const ethereum = windowObj.ethereum
+  const networkId = ethereum.chainId
+  const [status, setStatus] = useState(true)
+  const network = getNetworkFromChain(networkId)
+
+  useEffect(() => {
```

Comment:
On second thought, I think window.ethereum.on('networkChanged', ) is deprecated now, so might be ok to listen to changes to window.ethereum object (although it may lead to unnecessary RPC calls) if other keys in the ethereum object change.